doc/man/man3/crypto_poly1305.3monocypher

```diff
@@ -144,25 +145,22 @@ The attacker must not be able to guess the key;
 it must be shared with the recipient to allow verification;
 and it must be used
 .Sy only once .
-If it is ever reused, the attacker can easily recover it, then forge
-arbitrary messages, or even destroy all security.
+If it is ever reused, the attacker can easily recover it and then forge
+arbitrary messages.
 .Pp
-The session key cannot be used for this: it is secret and shared, but
-it is
-.Em reused .
-The attacker will recover it and break all security.
-A simple random number cannot be used either: there is no reasonable
-way to give it to the recipient without also revealing it to the
-attacker.
+Session keys cannot be used for this.
+They are shared and secret, but would be reused when sending multiple
+messages.
+Random numbers cannot be used either as there is no reasonable way to
+share it with the recipient without also revealing it to the attacker.
 .Pp
 The only practical source for the authentication key is a chunk of the
 encryption stream used to encrypt the message.
 That chunk must be
 .Em dedicated
```

Comment:
I don't understand this paragraph.

Does it go like this:

The "encryption stream" is like a random number stream seeded by the encryption key that you XOR with or whatever to encrypt the data. To authenticate a message you pull a 32 byte Poly1305 key from the head of the stream and do not use those bytes to encrypt any data (keyword in the text being "dedicated"), and then you encrypt your message with the rest of the stream

Reply:
I guess that is pretty much what it says in the next paragraph. maybe this one should be taken out

Reply:
Yes, that's exactly what it means.